src/sentry/incidents/models/alert_rule_activations.py

```diff
@@ -57,9 +59,9 @@ class AlertRuleActivations(Model):
 
     objects: ClassVar[AlertRuleActivationsManager] = AlertRuleActivationsManager()
 
-    alert_rule = FlexibleForeignKey("sentry.AlertRule", related_name="activations")
+    alert_rule = FlexibleForeignKey("sentry.AlertRule", related_name="activations", db_index=True)
     # date_added timestamp indicates when this particular run was activated
-    date_added = models.DateTimeField(default=timezone.now)
+    date_added = models.DateTimeField(default=timezone.now, db_index=True)
```

Comment:
You need a compound index here on `(alert_rule_id, date_added)`, not two individual indexes.